sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -3658,6 +3658,81 @@ pi_result piQueueGetInfo(pi_queue Queue, pi_queue_info ParamName,
   case PI_QUEUE_INFO_DEVICE_DEFAULT:
     die("PI_QUEUE_INFO_DEVICE_DEFAULT in piQueueGetInfo not implemented\n");
     break;
+  case PI_EXT_ONEAPI_QUEUE_INFO_STATUS: {
+    // We can exit early if we have in-order queue.
+    if (Queue->isInOrderQueue()) {
+      if (!Queue->LastCommandEvent)
+        return ReturnValue(pi_bool{true});
+
+      // We can check status of the event only if it isn't discarded otherwise
+      // it may be reset (because we are free to reuse such events) and
+      // zeEventQueryStatus will hang.
+      if (!Queue->LastCommandEvent->IsDiscarded) {
```

Comment:
Is there a more robust way to tell that `Queue->LastCommandEvent->ZeEvent` still represents the `Queue->LastCommandEvent`?

Reply:
Unfortunately, I don't see other way do this right now.

Reply:
Sounds like "dangling" pointers then. Can we put `ZeEvent` to nullptr in PI events that no longer represent them?

Reply:
Idea of putting ZeEvent to nullptr sounds good, but I wasn't able to change this easily, need to investigate this separately. If you don't mind I added TODO to follow up.

Reply:
OK